main/views.py

```diff
@@ -2,38 +2,33 @@
 main views
 """
 
+from pathlib import Path
+
+from django.conf import settings
 from django.http import (
+    HttpResponse,
     HttpResponseBadRequest,
     HttpResponseForbidden,
     HttpResponseNotFound,
 )
-from django.shortcuts import render
 from rest_framework.response import Response
 from rest_framework.viewsets import ViewSet
 
-from learning_resources.permissions import is_learning_path_editor
 from main.features import get_all_feature_flags, is_enabled
-from main.permissions import is_admin_user
 
 
 def index(request, **kwargs):  # pylint: disable=unused-argument  # noqa: ARG001
-    """Render the example app"""
-
-    user = request.user
-
-    js_settings = {
-        "user": {
-            "id": user.id,
-            "first_name": getattr(user, "first_name", None),
-            "last_name": getattr(user, "last_name", None),
-            "is_authenticated": bool(user.is_authenticated),
-            "is_learning_path_editor": user.is_authenticated
-            and (is_admin_user(request) or is_learning_path_editor(request)),
-            "is_article_editor": is_admin_user(request),
-        },
-    }
-
-    return render(request, "index.html", context={"js_settings": js_settings})
+    """
+    Return the static HTML index file for our react app.
+
+    In general, this should not be served by Django, but directly by nginx or
+    another web server.
+
+    However, if Django raises a 40x error, we want to serve this as a fallback
+    while preserving the erroring URL in the browser.
+    """
+    html_content = Path(f"{settings.STATIC_ROOT}/mit-open/index.html").read_text()
+    return HttpResponse(html_content)
```

Comment:
Is there a better way to serve a literally static HTML file? (I.e., not a template file). I could not find one.

Serving a purely static HTML file via django seems strange—we could just redirect—but I don't see any other way to preserve the URL (like http://localhost:8063/api/v1/cats-and-dogs).

Reply:
Any issue adding its path location to the Nginx config (as you suggested in the comments)? What's the content of the html file (to help me understand why we need to serve it at all)?

Reply:
HTML files are valid django templates - they just don't have and django template directives in them.

Since the files is under the static root directory, you'll probably need to add that directory to the templates directory configuration: https://docs.djangoproject.com/en/3.2/ref/settings/#dirs

Reply:
Given your other comment below, you probably want to put this directory first in the list and leave `main/templates/index.html` as the failover for tests.

Reply:
I ended up keeping the `touch ./frontends/mit-open/build/index.html` for tests. If the file is missing in prod, we probably want an error not a fallback.

The nginx `error_page` approach discussed in slack seems appealing, but I couldn't get it to work. It just always showed the django error_page.